irohad/consensus/yac/cluster_order.hpp

```diff
@@ -36,7 +36,7 @@ namespace iroha {
         /**
          * Provide current leader peer
          */
-        model::Peer currentLeader();
+        nonstd::optional<model::Peer> currentLeader();
```

Comment:
1. Why not boost?
2. Header seems missing

Reply:
Which header?

Reply:
for optional